apis/v1alpha1/route_types.go

```diff
@@ -0,0 +1,85 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// GatewayObjectReference identifies a Gateway object.
+type GatewayObjectReference struct {
+	// Namespace is the namespace of the referent.
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
+	// Name is the name of the referent.
+	//
+	// +kubebuilder:validation:Required
+	// +required
+	Name string `json:"name"`
+}
+
+// RouteConditionType is a type of condition for a route.
+type RouteConditionType string
+
+const (
+	// ConditionRouteAdmitted indicates whether the route has been admitted
+	// or rejected by a Gateway, and why.
+	ConditionRouteAdmitted RouteConditionType = "Admitted"
+)
+
+// RouteCondition is a status condition for a given route.
+type RouteCondition struct {
```

Comment:
Will this change to metav1.Condition once this merges? https://github.com/kubernetes-sigs/service-apis/pull/302

Reply:
Yes